submissions/Kostiantyn_Rakhmanov/oop-exercise/index.js

```diff
@@ -0,0 +1,75 @@
+import { print } from './js/lib.js';
+
+class Inhabitant {
+   constructor(species, name, gender, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.saying = saying;
+   }
+   printOut() {
+      return ['species', 'name', 'gender', 'saying'].map(
+         (prop) => `${prop}: ${this[prop]}`
+      ).join('; ');
+   }
+}
+
+class Animal extends Inhabitant {
+   constructor (species, name, gender, saying) {
+      super (species, name, gender, saying);
+      this.legs = 4;
```

Comment:
Why `name` is generalized to the base class and `legs` are not?
JFYI, methods are just functions, so their parameters can be assigned default values.

Reply:
Why do we need to reassign `legs` here?